src/main/kotlin/no/nav/familie/ef/sak/tilkjentytelse/TilkjentYtelseService.kt

```diff
@@ -27,6 +31,27 @@ class TilkjentYtelseService(private val behandlingService: BehandlingService,
                ?: error("Fant ikke tilkjent ytelse med behandlingsid $behandlingId")
     }
 
+    fun hentBeløpsperioderForBehandling(behandlingId: UUID): List<Beløpsperiode> {
+        val vedtakForBehandling = vedtakService.hentVedtak(behandlingId)
+
+        if (vedtakForBehandling.resultatType === ResultatType.OPPHØRT) {
+            throw Feil("Kan ikke vise fremtidige beløpsperioder for opphørt vedtak med id=$behandlingId")
+        }
+        val stønadstype = behandlingService.hentSaksbehandling(behandlingId).stønadstype
+
+        val startDatoForVedtak = when (stønadstype) {
+            StønadType.OVERGANGSSTØNAD -> vedtakForBehandling.perioder?.perioder?.minByOrNull { it.datoFra }?.datoFra
+            StønadType.BARNETILSYN -> vedtakForBehandling.barnetilsyn?.perioder?.minByOrNull { it.datoFra }?.datoFra
```

Comment:
Er det ikke BeløpsperiodeBarnetilsynDto vi ønsker å hente for barnetilsyn?

Reply:
Jo, det har du rett i :)